decoder.go

```diff
@@ -20,18 +60,56 @@ func frameLineSize(frame *C.AVFrame) *C.int {
 	return (*C.int)(unsafe.Pointer(&frame.linesize[0]))
 }
 
-// h264Decoder is a wrapper around ffmpeg's H264 decoder.
-type h264Decoder struct {
-	codecCtx    *C.AVCodecContext
-	srcFrame    *C.AVFrame
-	swsCtx      *C.struct_SwsContext
-	dstFrame    *C.AVFrame
-	dstFramePtr []uint8
+// getStreamInfo opens a stream URL and retrieves the video codec.
+func getStreamInfo(url string) (videoCodec, error) {
+	cUrl := C.CString(url)
+	defer C.free(unsafe.Pointer(cUrl))
```

Comment:
According to the docs:
https://pkg.go.dev/cmd/cgo
```
// Go string to C string
// The C string is allocated in the C heap using malloc.
// It is the caller's responsibility to arrange for it to be
// freed, such as by calling C.free (be sure to include stdlib.h
// if C.free is needed).
func C.CString(string) *C.char
```

I think we need to include `stdlib.h` so we can be sure `C.free` will succeed

Reply:
Interesting find! It seems as if the Go compiler is auto-"magically" adding stdlib - added it explicitly for good measure.